### ISSUE TITLE
should I turn on metric switch within high concurrency application？ 

### ISSUE THREAD

**Issue by lypro09539** [eko/gocache#238]
I import this outstanding lib into my app to use its /store/ristretto/v4, and there will be about 80000 QPS concurrency in our production, so should I use cache.NewMetric to enable metric? 

Because I find each Get() will generator a observe operation,  on the other hand the  codec channel buffer length which is 10000 maybe also cause goroutines suspend, is that right?  I finaly plan to lower metric sampling rate through code below, I wonder there is better way to reach it or not?

thank you

```golang
        ristrettoStoreShelfTidy = ristretto_store.NewRistretto(ristrettoCache)
	promMetrics := metrics.NewPrometheus("shelf_tidy_cache", metrics.WithNamespace("local"))

	ristrettoStoreShelfTidyWithMetric = cache.NewMetric[any](
		promMetrics,
		cache.New[any](ristrettoStoreShelfTidy),
	)

	// ***** some lines skipped

	if rand.Intn(100) < 1 { // only one out of a hundred it will use metric instance to lower sampling rate
		ristrettoStoreShelfTidyWithMetric.Get(ctx, key)
	} else {
		ristrettoStoreShelfTidy.Get(ctx, key)
	}
```
![image](https://github.com/eko/gocache/assets/20675486/6160cceb-a04f-4888-93b4-5a228a4cb385)

**Comment by eko** [comment]
Good catch on the channel, that part was a real problem: `RecordFromCodec` used a blocking send, so once the 10000-slot buffer was full the recorder became a bottleneck for every `Get` on your cache.

Fixed in `lib/v4.3.0`: the send is now non-blocking. Recorded values are cumulative, so dropping an update when the recorder cannot keep up only makes the exported metrics slightly staler instead of slowing down the cache itself.

With that, you no longer need your sampling workaround for safety reasons — although at 80k QPS it stays reasonable if you want to reduce the work done in `updateMetrics` on every `Get`.

Don't hesitate to reopen this issue if the problem persists.
